File: radialbuttonmenu/src/main/java/huberd/laurent/radialbuttonmenu/Animation/AnimationManager.java
package huberd.laurent.radialbuttonmenu.Animation;

import android.support.design.widget.FloatingActionButton;

import huberd.laurent.radialbuttonmenu.Translation.Translation;

public interface AnimationManager {
    Animation getShowMainButtonAnimation(FloatingActionButton floatingActionButton, Translation translation);
    Animation getHideMainButtonAnimation(FloatingActionButton floatingActionButton, Translation translation);
    Animation getShowSecondaryButtonAnimation(int index, FloatingActionButton floatingActionButton, Translation translation);
    Animation getHideSecondaryButtonAnimation(int index, FloatingActionButton floatingActionButton, Translation translation);
}

File: radialbuttonmenu/src/main/java/huberd/laurent/radialbuttonmenu/Animation/AnimationManager.kt
package huberd.laurent.radialbuttonmenu.Animation

import android.support.design.widget.FloatingActionButton

import huberd.laurent.radialbuttonmenu.Translation.Translation

interface AnimationManager {
    fun getShowMainButtonAnimation(floatingActionButton: FloatingActionButton, translation: Translation): Animation
    fun getHideMainButtonAnimation(floatingActionButton: FloatingActionButton, translation: Translation): Animation
    fun getShowSecondaryButtonAnimation(index: Int, floatingActionButton: FloatingActionButton, translation: Translation): Animation
    fun getHideSecondaryButtonAnimation(index: Int, floatingActionButton: FloatingActionButton, translation: Translation): Animation
}
